chain/chain/src/resharding.rs

```diff
@@ -275,15 +291,20 @@ impl Chain {
 
         let mut iter = get_flat_storage_iter(&store, shard_uid);
         while let Some(batch) = get_trie_update_batch(&mut iter) {
+            let TrieUpdateBatch { entries, size } = batch;
             // TODO(#9435): This is highly inefficient as for each key in the batch, we are parsing the account_id
             // A better way would be to use the boundary account to construct the from and to key range for flat storage iterator
             let (store_update, new_state_roots) = tries.add_values_to_split_states(
                 &state_roots,
-                batch,
+                entries,
                 &checked_account_id_to_shard_uid,
             )?;
             state_roots = new_state_roots;
             store_update.commit()?;
+            RESHARDING_BATCH_COUNT.with_label_values(&[shard_uid.to_string().as_str()]).inc();
+            RESHARDING_BATCH_SIZE
+                .with_label_values(&[shard_uid.to_string().as_str()])
+                .add(size as i64)
```

Comment:
We can try returning size as i64 instead of u64 in `get_trie_update_batch`?

Reply:
Either way we need to cast at some point. Logically size is unsigned so I feel better keeping it as so. Casting here seems ok because it is actually when it stops being "size" and becomes "arbitrary numeric value".